#  (markdown-links/no-self-destination)

> description

## ðŸ“– Rule Details

This rule reports ???.

<!-- eslint-skip -->

```md
<!-- eslint markdown-links/no-self-destination: 'error' -->

<!-- âœ“ GOOD -->


<!-- âœ— BAD -->

```

## ðŸ”§ Options

This rule has no options.

<!-- or -->

```json
{
  "markdown-links/no-self-destination": [
    "error",
    {}
  ]
}
```

- 

## ðŸ“š Further Reading

- 

## ðŸ‘« Related Rules

- [xxx]

[xxx]: https://xxx

