#  no-self-destination (markdown-links/no-self-destination)

> disallow redundant self-destination links

## ğŸ“– Rule Details

This rule reports redundant self-destination links. When a file links to itself using its own filename, a simple fragment should be used instead.

<!-- eslint-skip -->

```md
<!-- eslint markdown-links/no-self-destination: 'error' -->

<!-- âœ“ GOOD -->
[link](#fragment)
[link](./other.md#fragment)

<!-- âœ— BAD -->
[link](./self.md#fragment)
[link](self.md#fragment)
```

## ğŸ”§ Options

This rule has no options.

## ğŸ‘« Related Rules

- [markdown-links/no-missing-path](./no-missing-path.md)

